include/swift/AST/DiagnosticsCommon.def

```diff
@@ -54,6 +54,10 @@ NOTE(previous_decldef,none,
 NOTE(brace_stmt_suggest_do,none,
      "did you mean to use a 'do' statement?", ())
 
+WARNING(defer_stmt_at_block_end,none,
+     "'defer' at the end of its scope is redundant and will execute immediately; "
```

Comment:
Nit: The diagnostic definition is better placed within DiagnosticsSema.def as it's only emitted from Sema. DiagnosticsCommon is for common diagnostics that are emitted from separate parts of the compiler (e.g Parse and Sema).

Reply:
Ah, good catch. Yes, this belongs in `DiagnosticsSema.def`.

Reply:
thank you for catching this ... fixed in ffd13662d11fb1930fe4336cb8aef90d66175a35